cms/migrations/0027_populate_slug_values_faqs.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.3 on 2016-12-21 19:26
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from django.utils.text import slugify
+import itertools
+
+
+def gen_unique_slug(apps, schema_editor):
+    FrequentlyAskedQuestion = apps.get_model('cms', 'FrequentlyAskedQuestion')
+    for row in FrequentlyAskedQuestion.objects.all():
+        if not row.slug:
+            for x in itertools.count(1):
```

Comment:
Looks like you're trying to do an infinite loop, and break when you find a unique slug. The `itertools` module is great, but it's hiding your intent here. A `while` loop will more clearly express the intent here:

```python
if not row.slug:
    base_slug = slugify(row.question)
    slug = base_slug
    slug_is_unique = not FrequentlyAskedQuestion.objects.filter(slug=slug).exists()
    count = 1
    while not slug_is_unique:
        slug = "{base}-{count}".format(base=base_slug, count=count)
        slug_is_unique = not FrequentlyAskedQuestion.objects.filter(slug=slug).exists()
        count += 1
    row.slug = slug
    slug.save()
```